x/ibc/06-solomachine/types/client_state.go

```diff
@@ -0,0 +1,438 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	commitmentexported "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+)
+
+var _ clientexported.ClientState = ClientState{}
+
+// ClientState of a Solo Machine represents whether or not the client is frozen.
+type ClientState struct {
+	// Client ID
+	ID string `json:"id" yaml:"id"`
+
+	// Frozen status of the client
+	Frozen bool `json:"frozen" yaml:"frozen"`
+
+	// Current consensus state of the client
+	ConsensusState ConsensusState `json:"consensus_state" yaml:"consensus_state"`
```

Comment:
Should remove `ConsensusState` from here and replace with `LatestSequence uint64`. Whenever, you need to add a consensus state in this package, you should mock the behavior of the 02-client keeper.

```go
store.Set(host.KeyConsensusState(sequence), consState)
```

This will make sure that the getter/setter methods on the 02-client keeper works correctly for solo-machines, also necessary for misbehavior

Reply:
Hmm - we don't want to store the consensus state for each sequence, that's a large waste of space, and it's unnecessary since it can't be used to verify anything else. Is there another way to accomplish what we need for getter/setter methods and for misbehaviour here?

Reply:
this is not necessary because misbehaviour is now being properly handled with the LTE getter.